hail/src/main/scala/is/hail/types/physical/stypes/interfaces/SNDArray.scala

```diff
@@ -273,6 +333,159 @@ object SNDArray {
       C.firstDataAddress(cb), ldC)
   }
 
+  def trmm(cb: EmitCodeBuilder, side: String, uplo: String, transA: String, diag: String,
+    alpha: Code[Double], _A: SNDArrayCode, _B: SNDArrayCode): Unit = {
+    val A = _A.memoize(cb, "copy_A")
+    val B = _B.memoize(cb, "copy_B")
+    assertMatrix(A, B)
+    assertColMajor(cb, A, B)
+
+    val Seq(m, n) = B.shapes(cb)
+    val Seq(a0, a1) = A.shapes(cb)
+    cb.ifx(a1.cne(if (side == "left") m else n), cb._fatal("trmm: incompatible matrix dimensions"))
+    cb.ifx(a0 < a1, cb._fatal("trmm: A has fewer rows than cols"))
```

Comment:
If you include the row and col count in the message your life will be easier in the future.